Sources/Views/MessageLabel.swift

```diff
@@ -22,417 +22,491 @@
  SOFTWARE.
  */
 
-import UIKit
-
-open class MessageLabel: UILabel, UIGestureRecognizerDelegate {
-
-    // MARK: - Private Properties
-
-    private lazy var layoutManager: NSLayoutManager = {
-        let layoutManager = NSLayoutManager()
-        layoutManager.addTextContainer(self.textContainer)
-        return layoutManager
-    }()
-
-    private lazy var textContainer: NSTextContainer = {
-        let textContainer = NSTextContainer()
-        textContainer.lineFragmentPadding = 0
-        textContainer.maximumNumberOfLines = self.numberOfLines
-        textContainer.lineBreakMode = self.lineBreakMode
-        textContainer.size = self.bounds.size
-        return textContainer
-    }()
+// Missing UILabel Properties
 
-    private lazy var textStorage: NSTextStorage = {
-        let textStorage = NSTextStorage()
-        textStorage.addLayoutManager(self.layoutManager)
-        return textStorage
-    }()
+// adjustsFontSizeToFitWidth
+// allowsDefaultTighteningForTruncation
+// baselineAdjustment
+// minimumScaleFactor
+// highlightedTextColor
+// isHighlighted
+// shadowColor
+// shadowOffset
 
-    private lazy var rangesForDetectors: [DetectorType: [(NSRange, Any?)]] = [:]
-
-    // MARK: - Public Properties
+import UIKit
 
-    open weak var delegate: MessageLabelDelegate?
+extension NSTextStorage {
+    
+    var attributedString: NSAttributedString {
+        let range = NSRange(location: 0, length: mutableString.length)
+        return attributedSubstring(from: range)
+    }
+    
+    var mutableAttributedString: NSMutableAttributedString {
+        return NSMutableAttributedString(attributedString: attributedString)
+    }
+    
+    func addAttribute(key: NSAttributedStringKey, value: AnyObject, range: NSRange? = nil) {
+        let range = range ?? NSRange(location: 0, length: length)
+        addAttribute(key, value: value, range: range)
+    }
+    
+}
 
-    open var enabledDetectors: [DetectorType] = [.phoneNumber, .address, .date, .url]
+open class MessageLabel: UIView, UIGestureRecognizerDelegate {
 
-    open override var attributedText: NSAttributedString? {
-        didSet {
-            guard attributedText != oldValue else { return }
-            setTextStorage()
+    // MARK: - Properties [Public]
+    
+    public weak var delegate: MessageLabelDelegate?
+    
+    public var attributedText: NSAttributedString? {
+        get {
+            return _textSync
+        }
+        set {
+            _textSync = newValue
```

Comment:
It would be nice to strip these values and apply to the label properties still :/